homeassistant/components/roon/server.py

```diff
@@ -0,0 +1,156 @@
+"""Code to handle the api connection to a Roon server."""
+from asyncio import ensure_future, run_coroutine_threadsafe, sleep
+import logging
+
+from roon import RoonApi
+
+from homeassistant.const import CONF_API_KEY, CONF_HOST
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.util.dt import utcnow
+
+from .const import ROON_APPINFO
+
+_LOGGER = logging.getLogger(__name__)
+FULL_SYNC_INTERVAL = 30
+
+
+class RoonServer:
+    """Manages a single Roon Server."""
+
+    def __init__(self, hass, config_entry):
+        """Initialize the system."""
+        self.config_entry = config_entry
+        self.hass = hass
+        self.roonapi = None
+        self.all_player_ids = []
+        self.all_playlists = []
+        self.offline_devices = []
```

Comment:
Please make the containers that should hold unique items sets instead of lists. I suggest changing `all_player_ids` and `offline_devices`.